### DISCUSSION TITLE
Add Docker Support && Use GitHub Action to Cross Compile

### DISCUSSION THREAD

**Discussion by typenoob** [Brikwerk/nxbt#142] [Show and tell]
## Description

For those who have difficult in installing this application, I added dockerfile and built executable binary files in order to easily install on multiple platforms.

You can download executable files in [release](https://github.com/typenoob/nxbt/releases).

Requirements:
- Tag glib is for gnu based libc platform with glibc>=2.28
- Tag musl is for musl based libc platform with musl-libc>=1.2.2

Use following command to build your own image.

```docker build -t  [TAG_NAME] -f docker/[LIBC_KIND]/Dockerfile docker/gnu```

You can use built image from [here](https://hub.docker.com/r/typenoob/nxbt) as well.

**Comment by typenoob** [comment]
[Here is my fork](https://github.com/typenoob/nxbt).

**Comment by aneelx64** [comment]
Awesome. Any hints on the required environment variables. I'm assuming you have to expose access to the Bluetooth interface.

**Comment by AlbertoMonteiro** [comment]
Tyvm for that @typenoob 

With your releases and following those instructions https://github.com/dorssel/usbipd-win/discussions/310?sort=top#discussioncomment-2375886 I was able to make it work on windows via WSL 2 running Ubuntu 24

I didnt created my own kernel, I basically

- Followed the [WSL2 kernel build guide](https://github.com/dorssel/usbipd-win/wiki/WSL-support#building-your-own-usbip-enabled-wsl-2-kernel)
- Inserted modules:
  - sudo modprobe bluetooth
  - sudo modprobe btusb
  - sudo modprobe bcm203x (turns out was unnecessary, unloaded with sudo modprobe -r bcm203x)
- Started dbus and bluetooth services

After this I was able to use bluetoothctl successfully and nxbt started to work fine!!
